src/headless/converse-muc.js

```diff
@@ -369,9 +369,8 @@ converse.plugins.add('converse-muc', {
 
             extractReference (text, index) {
                 for (let i=index; i<text.length; i++) {
-                    if (text[i] !== '@') {
-                        continue
-                    } else {
+                    // Hack: refuse words like url with @ or email adresses.
+                    if (text[i] === '@' && (i <= 1 || text[i - 1] === ' ')) {
```

Comment:
This check is wrong, specifically `i <= 1`.

It should be `text[i] === '@' && (i === 0 || text[i-1] === ' ')`

Also, please remove the comment calling it a "hack", I don't think that's necessary.
A "hack" is something unexpected, ugly or temporary, and I don't think this change is that.